app/src/androidTest/java/com/github/geohunt/app/data/repository/FollowRepositoryTest.kt

```diff
@@ -0,0 +1,131 @@
+package com.github.geohunt.app.data.repository
+
+import com.github.geohunt.app.mocks.MockAuthRepository
+import com.github.geohunt.app.mocks.mockUser
+import com.github.geohunt.app.model.database.FirebaseEmulator
+import com.google.firebase.database.FirebaseDatabase
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.tasks.await
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.runTest
+import org.junit.After
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Test
+
+@Suppress("DEPRECATION")
+@OptIn(ExperimentalCoroutinesApi::class)
+class FollowRepositoryTest {
+    private lateinit var database: FirebaseDatabase
+    private val ioDispatcher = UnconfinedTestDispatcher()
+    private val auth = MockAuthRepository()
+    private val currentUserId = auth.getCurrentUser().id
+
+    @Before
+    fun setupEmulator() {
+        database = FirebaseEmulator.getEmulatedFirebase()
+    }
+
+    @After
+    fun clearEmulator() {
+        database.getReference("followList").removeValue()
+        database.getReference("followers").removeValue()
+    }
+
+    private fun followRepository(currentUser: AuthRepositoryInterface = auth): FollowRepositoryInterface {
+        return FollowRepository(currentUser, database, ioDispatcher)
+    }
+
+    @Test
+    fun followUserUpdatesDatabase() = runTest {
+        val follow = followRepository()
+
+        follow.follow(mockUser(id = "follow1"))
+
+        val follows = database.getReference("followList/$currentUserId/follow1").get().await().getValue(Boolean::class.java)
+        assertNotNull(follows)
+        assertTrue(follows!!)
```

Comment:
The `assertNotNull` is redundant here since `assertTrue` will fail on a `null` value.